docs/spatial-index-guide.md

```diff
@@ -0,0 +1,94 @@
+'INDEX_HANDLER'='mygeohash', 
+'INDEX_HANDLER.mygeohash.type'='geohash',   
+'INDEX_HANDLER.mygeohash.sourcecolumns'='longitude, latitude',   
+'INDEX_HANDLER.mygeohash.originLatitude'='19.832277',   
+'INDEX_HANDLER.mygeohash.gridSize'='50',   
+'INDEX_HANDLER.mygeohash.minLongitude'='1.811865',   
+'INDEX_HANDLER.mygeohash.maxLongitude'='2.782233',   
+'INDEX_HANDLER.mygeohash.minLatitude'='19.832277',   
+'INDEX_HANDLER.mygeohash.maxLatitude'='20.225281',   
+'INDEX_HANDLER.mygeohash.conversionRatio'='1000000');
+```
+
+| **Property**  | **Description**                                              |
+| ------------- | :----------------------------------------------------------- |
+| INDEX_HANDLER | Custom index handler. This handler allows user to  create a new column from the set of schema columns. Newly created column name  is same as that of handler name. Type and sourcecolumns properties for the handler are mandatory  properties. At present, only supported value for type property is 'geohash'. A simple  default implementation class can be available in carbon. User can hook their custom implementation class  for  geohash by extending the default implementation.     An  advanced sophisticated custom implementation of geohash can also be  made available in carbon. It can expect following additional table properties  for handler:  'INDEX_HANDLER.xxx.gridSize',  'INDEX_HANDLER.xxx.minLongitude',   'INDEX_HANDLER.xxx.maxLongitude',   'INDEX_HANDLER.xxx.minLatitude',   'INDEX_HANDLER.xxx.maxLatitude',     User can add their own table properties for  handler similar to above format and access them in their custom implementation  class. All those properties are optional at carbon. Can specify their implementation  class with 'INDEX_HANDLER.xxx.class' property  Default implementation class can support handler  column value generation taking source column values into account for each row  and support query filtering based on the source columns. Generated handler column  is invisible to user. Column is not allowed in any DDL commands and  properties except in SORT_COLUMNS table property. ' conversionRatio' allow user to translate Longitude and  Latitude to long. Example: when data loading the real Longitude=13.123456, Latitude=101.12356, configure INDEX_HANDLER.mygeohash.conversionRatio= '1000000', then user can change data to Longitude=13123456, Latitude=10112356. they are long type simpler and faster than floating-point number in calculation. |
+
+
+
+### Data Loading
+
+In this process, tables with spatial indexes configured will create invisible hash ids column data based on  'sourcecolumns' user defined and store it. Please ensure that the specified sourcecolumns column has data.
+
+### Data query
+
+CarbonData implements a UDF named 'IN_POLYGON'.
+
+example:
+
+```
+select * from source_index where IN_POLYGON('16.321011 4.123503,16.137676 5.947911,16.560993 5.935276,16.321011 4.123503'
```

Comment:
Please mention the minimum number of co-ordinates required for IN_POLYGON UDF.

Reply:
okay